_includes/gha-tutorial.md

```diff
@@ -0,0 +1,178 @@
+action!
+
+### Set up the workflow
+
+Set up your GitHub Actions workflow for building and pushing the image to Docker
+Hub.
+
+1. Go to your repository on GitHub and then select the **Actions** tab.
+2. Select **set up a workflow yourself**.
+
+   This takes you to a page for creating a new GitHub actions workflow file in
+   your repository, under `.github/workflows/main.yml` by default.
+
+3. In the editor window, copy and paste the following YAML configuration.
+
+   ```yaml
+   name: ci
+
+   on:
+     push:
+       branches:
+         - "main"
+
+   jobs:
+     build:
+       runs-on: ubuntu-latest
+   ```
+
+   - `name`: the name of this workflow.
+   - `on.push.branches`: specifies that this workflow should run on every push
+     event for the branches in the list.
+   - `jobs`: creates a job ID (`build`) and declares the type of machine that
+     the job should run on.
+
+For more information about the YAML syntax used here, see
+[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions){:
+target="blank" rel="noopener"}.
+
+### Define the workflow steps
+
+Now the essentials: what steps to run, and in what order to run them.
+
+The following YAML snippet contains a sequence of steps that:
+
+1. Checks out the repository on the build machine.
+2. Signs in to Docker Hub, using the
+   [Docker Login](https://github.com/marketplace/actions/docker-login){:
+   target="blank" rel="noopener"} action and your Docker Hub credentials.
+3. Creates a BuildKit builder instance using the
+   [Docker Setup Buildx](https://github.com/marketplace/actions/docker-setup-buildx){:
+   target="blank" rel="noopener"} action.
+4. Builds the container image and pushes it to the Docker Hub repository, using
+   [Build and push Docker images](https://github.com/marketplace/actions/build-and-push-docker-images){:
+   target="blank" rel="noopener"}.
+
+   The `with` key lists a number of input parameters that configures the step:
+
+   - `context`: the build context.
+   - `file`: filepath to the Dockerfile.
+   - `push`: tells the action to upload the image to a registry after building
+     it.
+   - `tags`: tags that specify where to push the image.
+
+{% raw %}
+
+```yaml
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          tags: ${{ secrets.DOCKER_HUB_USERNAME }}/clockbox:latest
+```
+
+{% endraw %}
+
+Add these steps to your workflow file. The full workflow configuration should
```

Comment:
Should this be a numbered step?

Reply:
I opted not to do this section as a series of ordered steps, just because I used it to describe the _workflow steps_, and this section only contains one user task.